cpp/src/rolling/rolling.cu

```diff
@@ -0,0 +1,260 @@
+		  void *in_col_data_ptr, gdf_valid_type *in_col_valid_ptr,
+		  gdf_size_type window,
+		  gdf_size_type min_periods,
+		  gdf_size_type forward_window,
+		  const gdf_size_type *window_col,
+		  const gdf_size_type *min_periods_col,
+		  const gdf_size_type *forward_window_col)
+  {
+    switch (agg_type) {
+    case GDF_SUM:
+      dispatch_aggregation_type<ColumnType, sum_op, false>(nrows, reinterpret_cast<ColumnType*>(out_col_data_ptr), out_col_valid_ptr, reinterpret_cast<ColumnType*>(in_col_data_ptr), in_col_valid_ptr, window, min_periods, forward_window, window_col, min_periods_col, forward_window_col);
+      break;
+    case GDF_MIN:
+      dispatch_aggregation_type<ColumnType, min_op, false>(nrows, reinterpret_cast<ColumnType*>(out_col_data_ptr), out_col_valid_ptr, reinterpret_cast<ColumnType*>(in_col_data_ptr), in_col_valid_ptr, window, min_periods, forward_window, window_col, min_periods_col, forward_window_col);
+      break;
+    case GDF_MAX:
+      dispatch_aggregation_type<ColumnType, max_op, false>(nrows, reinterpret_cast<ColumnType*>(out_col_data_ptr), out_col_valid_ptr, reinterpret_cast<ColumnType*>(in_col_data_ptr), in_col_valid_ptr, window, min_periods, forward_window, window_col, min_periods_col, forward_window_col);
+      break;
+    case GDF_COUNT:
+      dispatch_aggregation_type<ColumnType, count_op, false>(nrows, reinterpret_cast<ColumnType*>(out_col_data_ptr), out_col_valid_ptr, reinterpret_cast<ColumnType*>(in_col_data_ptr), in_col_valid_ptr, window, min_periods, forward_window, window_col, min_periods_col, forward_window_col);
+      break;
+    case GDF_AVG:
+      dispatch_aggregation_type<ColumnType, sum_op, true>(nrows, reinterpret_cast<ColumnType*>(out_col_data_ptr), out_col_valid_ptr, reinterpret_cast<ColumnType*>(in_col_data_ptr), in_col_valid_ptr, window, min_periods, forward_window, window_col, min_periods_col, forward_window_col);
+      break;
+    default:
+      // TODO: need a nice way to convert enums to strings, same would be useful for groupbys
+      CUDF_FAIL("Aggregation function " + std::to_string(agg_type) + " is not implemented");
+    }
+  }
+};
+
+}  // anonymous namespace
+
+namespace cudf {
+
+// see rolling.hpp for declaration
+gdf_column* rolling_window(const gdf_column *input_col,
+                           gdf_size_type window,
+                           gdf_size_type min_periods,
+                           gdf_size_type forward_window,
+                           gdf_agg_op agg_type,
+                           const gdf_size_type *window_col,
+                           const gdf_size_type *min_periods_col,
+                           const gdf_size_type *forward_window_col)
+{
+  // Make sure the input is not null
+  CUDF_EXPECTS(nullptr != input_col, "Input column pointer must not be null");
+
+  // Use the column wrapper class from io/utilities to quickly create a column
+  // TODO: should we care about naming the column?
```

Comment:
I don't think we ever worry about column names on the C++ side... I would leave it empty.